sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -326,47 +336,18 @@ function setName(entry: any, nameProperties: any): any {
  * @param response
  */
 export function buildError(errorBody: any, response: HttpOperationResponse): RestError {
-  const normalizedError: any = {};
-  const odataErrorFormat = !!errorBody["odata.error"];
-  const errorProperties =
-    errorBody.Error || errorBody.error || errorBody["odata.error"] || errorBody;
+  const errorProperties = errorBody.Error || errorBody.error || errorBody;
```

Comment:
How is it that we no longer treat `errorBody["odata.error"]` as the error object in case of odata?

Reply:
Since we don't use that information and it's already present on the `_response` ?

Reply:
In the code that was present before the assumption was that the `code` and `message` existed on `errorBody["odata.error"]` and not on `errorBody` itself when we are dealing with `odata`. With this change, we no longer look at `errorBody["odata.error"]`, but expect to find the `code` on `errorBody` directly.

Unless, in the case of `odata`, the `error.Error` or `error.error` is the one that is having the `code`.

I dont have knowledge of the expected error structure. I am going based off of the code that was added in #4922.

Reply:
I had run the tests, and the error details being picked are not ODATA ones, so that would go on to the `_response` body.
This is based on our previous discussion where we decided to not have to want to parse custom error properties and let them exist on the `_response` object. (unless we now want to do things differently)

Reply:
Ok, I think this is about the possibility that `Code` and `Detail` can exist as ODATA error properties. WHile this is not the case based on testing, updated to have it in just in case.

Reply:
> While this is not the case based on testing, updated to have it in just in case.

If we don't have a case where we expect the code/detail/message to be in `errorBody["odata.error"]`, then we shouldn't use it.

Reply:
We currently are not aware of any testcases that capture this. But given it existed in the original implementation, and going by similar documentation existing for another Azure service, https://docs.microsoft.com/en-us/rest/api/storageservices/status-and-error-codes2
It may be better/safe to have this in. Since it's the last boolean check/assignment, it wouldn't add any computational overhead in real time as well.
Any benefits or strong problems to having this in?

Reply:
If you want to allow the odata scenario, then shouldnt we be using the `message` property instead of the `detail` property which does not exist as per the spec you shared above?

Reply:
> Any benefits or strong problems to having this in?

The problem is in allowing a case of errors based on previous implementation but not having a scenario to test it.

The problem is also in allowing the case of odata errors, but not handling it as per the spec. Shouldnt we be using the `message` property instead of the `detail` property which does not exist as per the spec you shared above?

Reply:
>Shouldnt we be using the message property instead of the detail property which does not exist as per the spec you shared above?

The spec above is for Storage Service right? 
Per https://www.odata.org/documentation/odata-version-2-0/atom-format/ and other references, the property names seem to be custom ones dictated by implementation

Reply:
Removed it, can discuss with service team to look into scenarios where this may be important.

We should ideally do our best to find and surface relevant information to user where possible.

Reply:
Logged #6055 for further investigation on the matter